docs/API/components/form-builder.md

```diff
@@ -1,21 +1,80 @@
 # Fliplet Form Builder JS APIs
 
+## Retrieve an instance
+
+### `Fliplet.FormBuilder.get()`
+
+Retries the first or a specific form chartInstances
+
 ```js
 // the "get()" accepts the form name in case you have multiple forms on the page
-Fliplet.FormBuilder.get().then(function (form) {
-  // get the input value
-  form.field('foo').val();
+Fliplet.FormBuilder.get(name)
+  .then(function (form) {
+    // Use form to perform various actions
+  });
+```
+
+## Instance methods
+
+### `.field(name)`
+
+Retrieve a form field.
+
+```js
+Fliplet.FormBuilder.get()
+  .then(function (form) {
+    // Gets the field named 'foo'
+    // form.field('foo');
+  });
+```
+
+## Field methods
+
+### `.val()`
+
+Gets or sets the value of a field.
 
-  // set the input value
-  form.field('foo').val('bar');
+```js
+Fliplet.FormBuilder.get()
+  .then(function (form) {
+    // set the input value
+    form.field('foo').val('bar');
+
+    // gets the input value
+    return form.field('foo').val();
+  });
+```
 
-  // registers a callback to be fired whenever the field value changes
-  form.field('foo').change(function (val) {
+### `.change()`
 
-    // only show / enable this field when val is "bar"
-    // works like jQuery "toggle", which evaluates the argument as a "truthy" boolean.
-    form.field('barbaz').toggle(val === 'bar');
+Attaches event listeners to a field changed.
 
+```js
+Fliplet.FormBuilder.get()
+  .then(function (form) {
+    // registers a callback to be fired whenever the field value changes
+    form.field('foo').change(function (val) {
+    })
   });
-});
-```
\ No newline at end of file
+```
+
+### `.toggle()`
+
+Show and hide a field.
+
+```js
+Fliplet.FormBuilder.get()
+  .then(function (form) {
+    form.field('foo').toggle(); // Toggles field visibility
+    form.field('bar').toggle(true); // Shows field
+    form.field('baz').toggle(false); // Hides field
+  });
+```
+
+## Hooks
+
+### `beforeSubmit`
+
+```js
+Fliplet.Hooks.on('beforeFormSubmit')
```

Comment:
I would write something more like:
```js
Fliplet.Hooks.on('beforeFormSubmit', function(data) {
  // Your code here
});
```
For those that would copy and paste this and expect to work.